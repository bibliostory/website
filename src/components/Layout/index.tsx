import * as React from 'react';
import Helmet from 'react-helmet';
import styled, { ThemeProvider, createGlobalStyle } from 'styled-components';
import { useLayoutQuery } from '../../hooks/useLayoutQuery';
import theme from '../../utils/theme';
import { RouterProps } from '@reach/router';

const MainLayout = styled.main`
  max-width: 90%;
  margin: 1rem auto;
`;

const GlobalStyle = createGlobalStyle`
  body {
    background-color: ${({ theme }) => theme.colors.primary};
  }
`;

interface LayoutProps {
  children: React.ReactNode;
}

const Layout = ({ children }: LayoutProps & RouterProps) => {
  const { site } = useLayoutQuery();

  const { title, description, keywords } = site.siteMetadata;

  return (
    <ThemeProvider theme={theme}>
      <>
        <Helmet>
          <html lang="en" />
          <title>{title}</title>
          <meta name="description" content={description} />
          <meta name="keywords" content={keywords} />
        </Helmet>
        <GlobalStyle />
        <MainLayout>
          <div>{children}</div>
        </MainLayout>
      </>
    </ThemeProvider>
  );
};

export default Layout;
