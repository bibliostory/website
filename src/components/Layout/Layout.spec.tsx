import * as React from 'react';
import Layout from './';
import { create } from 'react-test-renderer';
import { useStaticQuery } from 'gatsby';

describe('<Layout />', () => {
  it('renders', () => {
    (useStaticQuery as jest.Mock).mockImplementation(() => ({
      site: {
        siteMetadata: {
          title: 'title',
          description: 'description',
          keywords: 'keywords',
        },
      },
    }));
    expect(create(<Layout>children</Layout>)).toMatchSnapshot();
  });
});
