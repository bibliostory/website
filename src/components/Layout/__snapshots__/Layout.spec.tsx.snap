// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Layout /> renders 1`] = `
<main
  className="sc-AykKC"
>
  <div>
    children
  </div>
</main>
`;
